audit.md

```diff
@@ -0,0 +1,212 @@
+- All unit tests passing (`aragonOS` only; see note above)
+- Uses industry-standard library contracts for common operations, including OpenZeppelin and MiniMe
+
+### Out of scope
+
+I have, for the most part, ignored:
+
+- Any contracts in `lib` folders
+- Any contracts in `factory` folders (they're mostly for demonstration right now)
+- An exhaustive look through the unit tests to see if they miss any obvious functionality (100% code and branch coverage is a good start but is usually not the whole story)
+
+
+## Issues
+
+- As a scale, I'm using:
+    - High-severity: anything that can result in loss of funds for users, or otherwise achieve highly damaging effects on user DAOs
+    - Mid-severity: exploitable issues, but either preventable in some way, technically difficult (or impossible) to solve for every one in the ecosytem, or very low probability of actually being exploited
+    - Low-severity: nice-to-haves, very low incentive attacks
+
+### Mid-severity
+
+- `AppProxy` should fail on construction if `_kernel` is not a contract
+    - `aragonOS/contracts/apps/AppProxy.sol`:
+        - Kernel-less apps are impossible: the fallback function will always fail on `kernel.getAppCode(appId)` if `kernel == 0` or if the kernel address is not a contract.
+    - `aragonOS/contracts/apps/AragonApp.sol`:
+        - `canPerform()` would not require the `address(kernel) == 0` check anymore as the proxy would fail to be created before that point
+    - `aragonOS/contracts/apm/ApmRegistry.sol`:
+        - `initialize()`'s `require(kernel != address(0));` check can be removed
+- `AppProxy` may use garbage data as its `target`:
+    - `aragonOS/contracts/apps/AppProxy.sol`:
+        - If the `AppProxy`'s `kernel` is set to a non-kernel contract address, `kernel.getAppCode(appId)` will return garbage data [due to how the fallback works](https://github.com/ethereum/solidity/issues/2630). This means that in `delegatedFwd(target, msg.data)`, `target` could potentially be delegating to a malicious contract. As far as I know, exploiting this would be extremely difficult, but it should be clear in the surrounding documentation that the `kernel` **MUST** be carefully set.
+- `KernelProxy` should fail on construction if `_kernelImpl` is not a contract
+    - `aragonOS/contracts/kernel/KernelProxy.sol`:
+        - `delegatedFwd(kernelImpl, msg.data)` will always fail if `kernelImpl == 0` or is not a contract. In such a case, it would also be impossible to run `upgradeKernel()`.
+- `KernelProxy`s' initialization can be front-runned:
+    - `aragonOS/contracts/kernel/KernelProxy.sol`:
+        - `initialize()` is public and accessible to anyone after the proxy is deployed. This should at least be highlighted in documentation for `Kernel.initialize()` and the `KernelProxy` so as to make sure users always create and initialize these proxies in one atomic transaction, but a better solution would be to introduce the same `initializePayload` that is in `AppProxy`.
```

Comment:
Yep, this is a documentation issue. DAOs should only be setup in factories in the EVM